libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFAuthenticationManager.m

```diff
@@ -348,35 +340,23 @@ - (void)dealloc
 - (BOOL)loginWithCompletion:(SFOAuthFlowSuccessCallbackBlock)completionBlock
                     failure:(SFOAuthFlowFailureCallbackBlock)failureBlock
 {
-    return [self loginWithCompletion:completionBlock failure:failureBlock account:nil];
+    return [self loginWithCompletion:completionBlock failure:failureBlock credentials:[[SFUserAccountManager sharedInstance] oauthCredentials]];
 }
 
+
 - (BOOL)loginWithCompletion:(SFOAuthFlowSuccessCallbackBlock)completionBlock
                     failure:(SFOAuthFlowFailureCallbackBlock)failureBlock
-                    account:(SFUserAccount *)account
+                    credentials:(SFOAuthCredentials *)credentials
 {
-    if (account == nil) {
-        account = [SFUserAccountManager sharedInstance].currentUser;
-        if (account == nil) {
-            // Create the current user out of legacy account data, if it exists.
-            account = [SFUserAccountManagerUpgrade createUserFromLegacyAccountData];
-            if (account == nil) {
-                [self log:SFLogLevelInfo format:@"No current user account, so creating a new one."];
-                account = [[SFUserAccountManager sharedInstance] createUserAccount];
-            }
-            [[SFUserAccountManager sharedInstance] saveAccounts:nil];
-        }
-    }
-    
+
     SFAuthBlockPair *blockPair = [[SFAuthBlockPair alloc] initWithSuccessBlock:completionBlock
                                                                   failureBlock:failureBlock];
     @synchronized (self.authBlockList) {
         if (!self.authenticating) {
             // Kick off (async) authentication.
             [self log:SFLogLevelDebug msg:@"No authentication in progress.  Initiating new authentication request."];
             [self.authBlockList addObject:blockPair];
-            [self loginWithUser:account];
-            
+            [self loginWithCredentials:credentials];
             return YES;
         } else {
             // Already authenticating.  Add completion blocks to the list, if they're not there already.
```

Comment:
Even if we don't do it in this PR (which I think is probably okay), we should add a work item to consider the fact that there needs to be some bounds checking on "piggy-backing" authentication requests that may not be related.  For instance, it would definitely be imprudent to piggy-back one user's credentials refresh on an in-flight refresh of another user's credentials, or a refresh on an in-flight new authentication, etc.

Probably outside of the scope of this review, but I think worth capturing.